tests/test_pylint_datetime.py

```diff
@@ -2,7 +2,308 @@
+        )
+        self.asserting_messages_assign(
+            "naive-datetime-no-timezone", call_node_2, args=("max",)
+        )
+        self.asserting_messages_assign(
+            "naive-datetime-no-timezone", call_node_3, args=("fromisocalendar",)
+        )
+        self.asserting_messages_assign(
+            "naive-datetime-no-timezone", call_node_4, args=("fromordinal",)
+        )
+        self.asserting_messages_assign(
+            "naive-datetime-no-timezone", call_node_5, args=("min",)
+        )
+        self.asserting_messages_assign(
+            "naive-datetime-no-timezone", call_node_6, args=("max",)
+        )
+
+    def test_naive_objects_falsely_replaced(self) -> None:
+        """Test naive objects with subsequent call to replace but lacking timezone info."""
+        (
+            call_node_1,
+            call_node_2,
+            call_node_3,
+            call_node_4,
+            call_node_5,
+            call_node_6,
+            call_node_7,
+            call_node_8,
+            call_node_9,
+            call_node_10,
+            call_node_11,
+            call_node_12,
+        ) = astroid.extract_node(
+            """
+        import datetime
+        dt_min = datetime.datetime.min #@
+        dt_min.replace(1) #@
+        dt_max = datetime.datetime.max #@
+        dt_max.replace(1) #@
+        dt1 = datetime.datetime.fromisocalendar(1, 1, 1) #@
+        dt1.replace(1) #@
+        dt2 = datetime.datetime.fromordinal(1) #@
+        dt2.replace(year=1) #@
+        t_min = datetime.time.min #@
+        t_min.replace(year=1) #@
+        t_max = datetime.time.max #@
+        t_max.replace(year=1) #@
+
+        """
+        )
+        self.asserting_messages_assign(
+            "timezone-no-argument", call_node_1, call_node_2, args=("min",)
+        )
+        self.asserting_messages_assign(
+            "timezone-no-argument", call_node_3, call_node_4, args=("max",)
+        )
+        self.asserting_messages_assign(
+            "timezone-no-argument", call_node_5, call_node_6, args=("fromisocalendar",)
+        )
+        self.asserting_messages_assign(
+            "timezone-no-argument", call_node_7, call_node_8, args=("fromordinal",)
+        )
+        self.asserting_messages_assign(
+            "timezone-no-argument", call_node_9, call_node_10, args=("min",)
+        )
+        self.asserting_messages_assign(
+            "timezone-no-argument", call_node_11, call_node_12, args=("max",)
+        )
+
+    def test_naive_objects_correctly_replaced(self) -> None:
+        """Test naive objects with subsequent call to replace including timezone info."""
+        (
+            call_node_1,
+            call_node_2,
+            call_node_3,
+            call_node_4,
+            call_node_5,
+            call_node_6,
+        ) = astroid.extract_node(
+            """
+        import datetime
+        dt_min = datetime.datetime.min #@
+        dt_min.replace(tzinfo=1)
```

Comment:
Is there any reason to call these `.replace` after the assignment?

Reply:
I don't think so.

Reply:
actually that may be a case I didn't think of, because I wasn't aware of the usage of .replace